go/tools/builders/nogo_fix.go

```diff
@@ -0,0 +1,241 @@
+
+	return out, nil
+}
+
+// getFixes merges the suggested fixes from all analyzers, returns one fileChange object per file,
+// while reporting conflicts as error.
+func getFixes(entries []diagnosticEntry, fileSet *token.FileSet) ([]fileChange, error) {
+	var allErrors []error
+	finalChanges := make(map[string][]nogoEdit)
+
+	for _, entry := range entries {
+		// According to the [doc](https://pkg.go.dev/golang.org/x/tools@v0.28.0/go/analysis#Diagnostic),
+		// an analyzer may suggest several alternative fixes, but only one should be applied.
+		// We will go over all the suggested fixes until the we find one with no conflict
+		// with previously selected fixes. No backtracking is used for simplicity and performance. If
+		// none of the suggested fixes of a diagnostic can be applied, the diagnostic entry will be skipped
+		// with an error message to the user.
+		foundApplicableFix := false
+		for _, sf := range entry.Diagnostic.SuggestedFixes {
+			candidateChanges := make(map[string][]nogoEdit)
+			applicable := true
+			for _, edit := range sf.TextEdits {
+				start, end := edit.Pos, edit.End
+				if !end.IsValid() {
+					end = start
+				}
+
+				file := fileSet.File(start)
+				if file == nil {
+					// missing file info, most likely due to analyzer bug.
+					applicable = false
+					break
+				}
+
+				fix := nogoEdit{
+					Start: file.Offset(start),
+					End: file.Offset(end),
+					New: string(edit.NewText),
+					analyzerName: entry.analyzerName,
+				}
+				candidateChanges[file.Name()] = append(candidateChanges[file.Name()], fix)
+			}
+			// validating the edits from current SuggestedFix. All edits from a SuggestedFix must be
+			// either accepted or discarded atomically, because a SuggestedFix may move a statement from one place
+			// to the other. If we only accept part of the edits, the statement may either appear twice or disappear.
+			for fileName, edits := range candidateChanges {
+				edits = append(edits, finalChanges[fileName]...)
+				var err error
+				if candidateChanges[fileName], err = validate(edits); err != nil {
+					applicable = false
+					break
+				}
+			}
+			if applicable {
+				for fileName, edits := range candidateChanges {
+					finalChanges[fileName] = edits
+				}
+				foundApplicableFix = true
+				break
+			}
+			// Move on to the next SuggestedFix of the same Diagnostic if any edit of the current SuggestedFix has issues.
+		}
+		if !foundApplicableFix {
+			allErrors = append(allErrors, fmt.Errorf(
+				"ignoring suggested fixes from analyzer %q at %s",
+				entry.analyzerName, fileSet.Position(entry.Pos),
+			))
+		}
+	}
+
+	var finalFileChanges []fileChange
+	for fileName, edits := range finalChanges {
+		finalFileChanges = append(finalFileChanges, fileChange{fileName: fileName, changes: edits})
+	}
+
+	if len(allErrors) == 0 {
+		return finalFileChanges, nil
+	}
+
+	var errMsg bytes.Buffer
+	errMsg.WriteString("some suggested fixes are invalid or have conflicts with other fixes:")
```

Comment:
Please add a suggested follow-up such as applying other fixes and rerunning the build.